pom.xml

```diff
@@ -170,12 +170,8 @@
     <scala.version>2.12.17</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
     <scalatest-maven-plugin.version>2.2.0</scalatest-maven-plugin.version>
-    <!--
-      This needs to be managed in different profiles to avoid
-      errors building different Hadoop versions.
-      See: SPARK-36547, SPARK-38394.
-       -->
-    <scala-maven-plugin.version>4.8.1</scala-maven-plugin.version>
+    <!-- dont update scala-maven-plugin to version 4.8.1 SPARK-42809 and SPARK-43595 -->   
+    <scala-maven-plugin.version>4.8.0</scala-maven-plugin.version>
```

Comment:
```suggestion
    <!--
      This needs to be managed in different profiles to avoid
      errors building different Hadoop versions.
      See: SPARK-36547, SPARK-38394.
       -->
    <scala-maven-plugin.version>4.8.1</scala-maven-plugin.version>
```